src/Parser/Core/Modules/SpellUsable.js

```diff
@@ -0,0 +1,247 @@
+import SPELLS from 'common/SPELLS';
+import { formatDuration } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+
+import CastEfficiency from './CastEfficiency';
+
+const debug = false;
+
+function spellName(spellId) {
+  return SPELLS[spellId] ? SPELLS[spellId].name : '???';
+}
+
+class SpellUsable extends Module {
+  static dependencies = {
+    castEfficiency: CastEfficiency,
+  };
+  _currentCooldowns = {};
+
+  /**
+   * Whether the spell can be cast. This is not the opposite of `isOnCooldown`! A spell with 2 charges, 1 available and 1 on cooldown would be both available and on cooldown at the same time.
+   */
+  isAvailable(spellId) {
+    if (this.isOnCooldown(spellId)) {
+      const maxCharges = this.castEfficiency.getMaxCharges(spellId);
+      if (maxCharges > this._currentCooldowns[spellId].chargesOnCooldown) {
+        return true;
+      }
+      return false;
+    } else {
```

Comment:
This else can be removed.

Reply:
I prefer it this way, since it's a taste thing I'll leave it as it is 😜